mesonbuild/interpreter/type_checking.py

```diff
@@ -347,3 +380,59 @@ def _output_validator(outputs: T.List[str]) -> T.Optional[str]:
     validator=in_set_validator({'set', 'prepend', 'append'}))
 
 ENV_SEPARATOR_KW = KwargInfo('separator', str, default=os.pathsep)
+
+DEPENDENCIES_KW: KwargInfo[T.List[Dependency]] = KwargInfo(
+    'dependencies',
+    ContainerTypeInfo(list, (Dependency, ExternalLibrary, InternalDependency)),
```

Comment:
`ExternalLibrary` and `InternalDependency` are both `Dependency` subclasses, so it doesn't seem useful to list them here?

Reply:
This code predates d413dedf2a6a6f893e110f7a03d9e3468fdf1fc3 and was never modified to sync with the exact types from interpreter.py

On the other hand, KwargInfo will actually use this information when emitting an error message stating the types you are allowed to use. So, I don't know. Maybe it makes sense to list them all? Maybe it doesn't?

Reply:
Yeah, I guess leave them, or at least leave ExternalLibrary which is not an obvious `Dependency` type. But probably add a comment to explain why we're leaving them.